### PR TITLE
Use FormatterKit to display short dates.

### LINKED ISSUE #5217: Date formats are not consistent inside the app for the Reader and Post List and Calypso
Today I was editing some posts using the native iOS app and using Calypso on the Web and I noticed some discrepancies  between them on the displaying of post dates.

Here are the web versions:

screen shot 2016-04-21 at 13 02 13

and here are the iOS versions:
screen shot 2016-04-21 at 13 07 55

On the IOS app it looks we write the minutes passed (5m) and then hours passed (1h) and the fall back to days passed (20d) and this can go to whatever number (1000d).
On Calypso on the Reader it looks that after 6 days passed (6d ago) we fall back to the post actual date (April 13, 2016) but on the Post list we go up to 30 days and then fall back to months passed (2 months ago) and years passed ( 2 years ago)

Should't we make all of these dates formats consistent?

cc @mattmiklic

### PR DESCRIPTION
**Pull request by SergioEstevao**
Fixes #5217 

To test: 

 - Go to the reader and scroll around and check the date formats being displayed.
 - Go to post list and check the dates on the cells.
 - Go to pages list and check the dates on the headers

If possible switch to another language that you know and it should be supported for the app and see if the translation shows up

Needs review: @aerych

### CODE REVIEW

**Comment by astralbodies** [comment]
@SergioEstevao Can you check in the updated Podfile.lock - Gridicons is causing the build to fail with Travis. 😞

**Comment by koke** [comment]
Did you check scrolling performance? I think one of the reasons we had the old code is because a more complex relative date was too slow

**Comment by koke** [comment]
After scrolling for a while on the Reader on an iPad Mini:

screen shot 2016-04-26 at 16 28 50

I'd say 👍 on the performance front 😄

**Comment by aerych** [comment]
Hiya @SergioEstevao !

Loving this change for the post list. Dates there are much more consistent with the web. 

This change stays consistent with the reader for about the first seven days, but then on the web the date format shifts to a short month day year format. Example: Apr 18, 2016. The "kanji" tag is a good one to test this with, or any other tag that gets a moderate amount of traffic. 

Could we tweak this PR to use the alternate format in the reader after the first week?
screen shot 2016-04-26 at 10 33 51 am

**Comment by SergioEstevao** [comment]
@aerych and @astralbodies it's ready for another look.

**Comment by aerych** [comment]
:shipit: from me once the conflicts are resolved :)

**Comment by astralbodies** [comment]
Reviewing the changes now...

**Comment by astralbodies** [comment]
I think you're missing the locale parameter on the NSDateFormatter -

File: WordPress/Classes/Categories/NSDate+StringFormatting.m [hunk 1]
```diff
@@ -1,8 +1,22 @@
...
+#import <FormatterKit/TTTTimeIntervalFormatter.h>
 
 @implementation NSDate (StringFormatting)
 
++ (NSDateFormatter *)shortDateFormatter
+{
+    static NSDateFormatter *_shortDateFormatter = nil;
+    static dispatch_once_t _onceToken;
+    dispatch_once(&_onceToken, ^{
+        _shortDateFormatter = [[NSDateFormatter alloc] init];
```
Comment (astralbodies) [inline WordPress/Classes/Categories/NSDate+StringFormatting.m]: Probably need to set a locale here to resolve the issue I commented on the PR about.

**Comment by astralbodies** [comment]
Weird. It looks like `currentLocale` and `autoupdatingCurrentLocale` aren't getting overriden properly for me with the scheme launch parameters. Stabby.

:shipit: after the conflict is resolved.

**Pull request merged**